src/textual/widgets/_markdown.py

```diff
@@ -500,19 +501,35 @@ class MarkdownFence(MarkdownBlock):
     def __init__(self, markdown: Markdown, code: str, lexer: str) -> None:
         self.code = code
         self.lexer = lexer
+        self.theme = "solarized-dark" if self.app.dark else "solarized-light"
         super().__init__(markdown)
 
-    def compose(self) -> ComposeResult:
-        from rich.syntax import Syntax
+    def _retheme(self) -> None:
+        """Swap between a dark and light theme when the mode changes."""
+        self.theme = "solarized-dark" if self.app.dark else "solarized-light"
```

Comment:
If the names of the preferences for the light/dark themes move up to `Markdown`, this might be moot, but I feel that hard-coding the theme choices in two different places is something we should avoid.